skcosmo/feature_selection/_greedy.py

```diff
@@ -0,0 +1,154 @@
+"""
+Sequential feature selection
+"""
+import numbers
+
+import numpy as np
+
+from sklearn.feature_selection._base import SelectorMixin
+from sklearn.base import BaseEstimator, MetaEstimatorMixin
+from sklearn.utils.validation import check_is_fitted, check_array
+
+
+class GreedySelector(SelectorMixin, MetaEstimatorMixin, BaseEstimator):
+    """Transformer that performs Greedy Feature Selection.
+
+    This Greedy Selector adds (forward selection) features to form a
+    feature subset in a greedy fashion. At each stage, the model scores each
+    feature (without an estimator) and chooses the feature with the maximum score.
+
+    Parameters
+    ----------
+
+    n_features_to_select : int or float, default=None
+        The number of features to select. If `None`, half of the features are
+        selected. If integer, the parameter is the absolute number of features
+        to select. If float between 0 and 1, it is the fraction of features to
+        select.
+
+    scoring : str, callable, list/tuple or dict, default=None
+        A single str (see :ref:`scoring_parameter`) or a callable
+        (see :ref:`scoring`) to evaluate the features. It is assumed that the
+        next feature to select is that which maximizes the score.
+
+        NOTE that when using custom scorers, each scorer should return a single
+        value. Metric functions returning a list/array of values can be wrapped
+        into multiple scorers that return one value each.
+
+    Attributes
+    ----------
+    n_features_to_select_ : int
+        The number of features that were selected.
+
+    support_ : ndarray of shape (n_features,), dtype=bool
+        The mask of selected features.
+
+    """
+
+    def __init__(
+        self,
+        scoring,
+        n_features_to_select=None,
+    ):
+
+        self.n_features_to_select = n_features_to_select
+        self.scoring = scoring
+
+    def fit(self, X, y=None, current_mask=None):
+        """Learn the features to select.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training vectors.
+        y : array-like of shape (n_samples,)
+            Target values.
+        current_mask : array-like of shape (n_features)
```

Comment:
Oh, ok, so you think that it is better to pass on the mask as an argument to `fit`, not to the `__init__`?

Reply:
I think so -- although I'm willing to debate.

Reply:
On the whole I agree too, yes. Especially since we are passing all the parameters related to a specific task into the `fit`. But in my opinion, in that case it makes more sense to move the `n_features_to_select` parameter to `fit` too. Or is there a slightly different approach here?

Reply:
the n_features_to_select is a constructor argument in sklearn feature selectors, hence the designation here

Reply:
Hm, ok. It's just a bit strange to me that it's n_features_to_select that is selected at the beginning and can't be changed. And I don't really understand why sklearn chose this format.

Reply:
I agree, it's counterintuitive, but I do see some of the reasoning behind the format. The idea is that the selector should have static meta-variables (i.e. _how_ the selection is chosen) and can be re-fit with new data. Therefore, anything that is regarding the process, not the exact data itself, should be passed to the constructor, and the exact data should be passed to the `fit`.

Reply:
Ok, I catch the idea, agree with your idea